src/mcmc/Inference.jl

```diff
@@ -78,30 +78,75 @@ abstract type Hamiltonian <: InferenceAlgorithm end
 abstract type StaticHamiltonian <: Hamiltonian end
 abstract type AdaptiveHamiltonian <: Hamiltonian end
 
+_bool_from_val(::Val{true}) = true
+_bool_from_val(::Val{false}) = false
+_bool_from_val(@nospecialize(x)) = throw(ArgumentError("Expected Val{true} or Val{false}, got $x"))
+
 """
-    ExternalSampler{S<:AbstractSampler}
+    ExternalSampler{S<:AbstractSampler,AD<:ADTypes.AbstractADType,Unconstrained}
+
+Represents a sampler that is not an implementation of `InferenceAlgorithm`.
+
+The `Unconstrained` type-parameter is to indicate whether the sampler requires unconstrained space.
 
 # Fields
 $(TYPEDFIELDS)
 """
-struct ExternalSampler{S<:AbstractSampler} <: InferenceAlgorithm
+struct ExternalSampler{S<:AbstractSampler,AD<:ADTypes.AbstractADType,Unconstrained} <: InferenceAlgorithm
     "the sampler to wrap"
     sampler::S
+    "the automatic differentiation (AD) backend to use"
+    adtype::AD
+
+    """
+        ExternalSampler(sampler::AbstractSampler, adtype::ADTypes.AbstractADType, ::Val{unconstrained})
+
+    Wrap a sampler so it can be used as an inference algorithm.
+
+    # Arguments
+    - `sampler::AbstractSampler`: The sampler to wrap.
+    - `adtype::ADTypes.AbstractADType`: The automatic differentiation (AD) backend to use.
+    - `unconstrained::Val=Val{true}()`: Value type containing a boolean indicating whether the sampler requires unconstrained space.
+    """
+    function ExternalSampler(
+        sampler::AbstractSampler,
+        adtype::ADTypes.AbstractADType,
+        unconstrained::Union{Val{false},Val{true}}=Val{true}()
+    )
+        return new{typeof(sampler),typeof(adtype),_bool_from_val(unconstrained)}(sampler, adtype)
```

Comment:
Wouldn't it be easier to write
```suggestion
       ::Val{unconstrained}=Val(true)
    ) where {unconstrained}
        if !(unconstrained isa Bool)
            throw(ArgumentError("Expected Val{true} or Val{false}, got Val{$unconstrained}"))
        end
        return new{typeof(sampler),typeof(adtype),unconstrained}(sampler, adtype)
```